src/metabase/driver/sql_jdbc/actions.clj

```diff
@@ -539,28 +568,145 @@
                     {:field-name->id field-name->id, :row row, :status-code 400})))
   (into [:and] (for [[field-name value] row
-                     :let               [field-id (get field-name->id (u/qualified-name field-name))
-                                         ;; if the field isn't in `field-name->id` then it's an error in our code. Not
-                                         ;; i18n'ed because this is not something that should be User facing unless our
-                                         ;; backend code is broken.
-                                         ;;
-                                         ;; Unknown column names in user input WILL NOT trigger this error.
-                                         ;; [[row->mbql-filter-clause]] is only used for *known* PK columns that are
-                                         ;; used for the MBQL `:filter` clause. Unknown columns will trigger an error in
-                                         ;; the DW but not here.
+                     :let               [field-id (get field-name->id field-name)
+                                        ;; if the field isn't in `field-name->id` then it's an error in our code. Not
+                                        ;; i18n'ed because this is not something that should be User facing unless our
+                                        ;; backend code is broken.
+                                        ;;
+                                        ;; Unknown column names in user input WILL NOT trigger this error.
+                                        ;; [[row->mbql-filter-clause]] is only used for *known* PK columns that are
+                                        ;; used for the MBQL `:filter` clause. Unknown columns will trigger an error in
+                                        ;; the DW but not here.
                                          _ (assert field-id
                                                    (format "Field %s is not present in field-name->id map"
                                                            (pr-str field-name)))]]
                  [:= [:field field-id nil] value])))
 
+;;;; Foreign Key Cascade Deletion
+
+(mu/defn- build-fk-metadata :- [:map-of ::lib.schema.id/table [:sequential :map]]
+  "Build foreign key relationship metadata for a database.
+  Returns a map of parent-table-id -> [{:table child-table-id, :fk {child-fk-col parent-pk-col}, :pk [child-pk-cols]}]"
+  [database-id :- ::lib.schema.id/database]
+  (actions/cached-value
+   [::build-fk-metadata database-id]
+   (fn []
+     (qp.store/with-metadata-provider database-id
+       (let [all-tables          (lib.metadata.protocols/tables (qp.store/metadata-provider))
+             all-fields          (mapcat #(lib.metadata.protocols/fields (qp.store/metadata-provider) (:id %)) all-tables)
+             fk-fields           (filter #(and (:fk-target-field-id %) (:active %)) all-fields)
```

Comment:
If you want to be 😎 then you can use `(every-pred :fk-target-field-id :active)`

Reply:
coollll